r-package/grf/tests/testthat/test_analysis_tools.R

```diff
@@ -38,8 +38,37 @@ test_that("leaf samples are indexed correctly", {
 
 	samples = unlist(sapply(quantile.tree$nodes, function(node) node$samples))
 	expect_equal(seq(1, n), sort(samples))
+
+})
+
+test_that("draw samples are indexed correctly",{
+  p = 40
```

Comment:
Instead of this check, I think we should use the logic from the above test case "leaf samples are indexed correctly". It will be more consistent, and is also more of a concise + direct test.